src/python/pants/engine/build_files.py

```diff
@@ -80,6 +81,17 @@ def _raise_did_you_mean(address_family: AddressFamily, name: str, source=None) -
     raise resolve_error from source
   raise resolve_error
 
+@rule
+async def find_build_file(address: Address) -> BuildFileAddress:
```

Comment:
So, this works, but we should take a look at how it is being used, and see whether it should be batched. The parsing will be memoized obviously, but each usage is O(N) in the number of targets in a directory.

Reply:
Sounds good. For now, it adds no new complexity than what we had before. But, after we finish this project, we'll be able to see if we can improve this. Thanks!